tests/syscall_module/test

```diff
@@ -37,35 +37,52 @@ system("auditctl -D >& /dev/null");
 
 # set the socketcall filter
 my $key = key_gen();
+my $result;
 # connect
-system("auditctl -a always,exit -F arch=b64 -S init_module -S finit_module -k $key");
+system("auditctl -a always,exit -F arch=b64 -S init_module -S finit_module -k $key-load");
+system("auditctl -a always,exit -F arch=b64 -S delete_module -k $key-unload");
 
 # run the test
-my $name = "arptable_filter";
-system("modprobe $name");
-
-system("sleep 5;sync");
+my $name = "arp_tables";
+$result = system("modprobe $name >/dev/null 2>&1");
+ok($result, 0); # Did the modprobe succeed?
+system("sleep 1;sync");
 
 # test if we generate any audit records from the filter rule
-my $result = system("ausearch -k $key > $stdout 2> $stderr");
-ok($result, 0);
+$result = system("ausearch -k $key-load > $stdout 2> $stderr");
+ok($result, 0); # Did the load search find records?
 
 # test if we generate the MODULE record
 my $line;
-my $line2;
 my $type;
 my $found_name = 0;
 while ($line = <$fh_out>) {
-	if ($line =~ /^type=UNKNOWN\[1330\] /) {
+	if ($line =~ /^type=KERN_MODULE /) {
+		if ($line =~ / name=\"$name\"/) {
+			$found_name = 1;
+		}
+	}
+}
+ok($found_name); # Was the load module found?
+
+$result = system("rmmod $name");
+ok($result, 0); # Did the rmmod succeed?
+system("sleep 1;sync");
+seek($fh_out, 0, 0);
+seek($fh_err, 0, 0);
+$found_name = 0;
+$result = system("ausearch -k $key-unload > $stdout 2> $stderr");
+ok($result, 0); # Did the unlaod search find records?
```

Comment:
"unlaod"? ;)

Don't worry about it, this looks fine to me otherwise, I can fix this while merging.